src/service/viewer-impl.js

```diff
@@ -872,8 +872,16 @@ export class Viewer {
       return Promise.resolve('');
     }
     return this.sendMessageUnreliable_('fragment', undefined, true).then(
-      hash => hash || ''
-    );
+        hash => {
+          if (!hash) {
+            return '';
+          }
+          /* Strip leading '#' */
+          if (hash.indexOf('#') == 0) {
```

Comment:
@jridgewell @cramforce Since in the `updateFragment` message I am planning to provide to the viewer is no-leading-`#` fragment, maybe it's better to keep it consistent: no-leading-`#` in the communication with viewer. We can throw error if we receive leading-`#` fragment so that it's not a silent failure.